data/account/participation.go

```diff
@@ -176,23 +190,39 @@ func FillDBWithParticipationKeys(store db.Accessor, address basics.Address, firs
 	// Generate them
 	v := crypto.GenerateOneTimeSignatureSecrets(firstID.Batch, numBatches)
 
-	// Also generate a new VRF key, which lives in the participation keys db
+	// Generate a new VRF key, which lives in the participation keys db
 	vrf := crypto.GenerateVRFSecrets()
 
+	// TODO change this
+	compactCertRound := config.Consensus[protocol.ConsensusFuture].CompactCertRounds
+
+	// Generate a new key which signs the compact certificates
+	stateProofSecrets, err := merklekeystore.New(uint64(firstValid), uint64(lastValid), compactCertRound, crypto.DilithiumType, store)
+	if err != nil {
+		return PersistedParticipation{}, err
+	}
+
 	// Construct the Participation containing these keys to be persisted
 	part = PersistedParticipation{
 		Participation: Participation{
-			Parent:      address,
-			VRF:         vrf,
-			Voting:      v,
-			FirstValid:  firstValid,
-			LastValid:   lastValid,
-			KeyDilution: keyDilution,
+			Parent:            address,
+			VRF:               vrf,
+			Voting:            v,
+			StateProofSecrets: stateProofSecrets,
+			FirstValid:        firstValid,
+			LastValid:         lastValid,
+			KeyDilution:       keyDilution,
 		},
 		Store: store,
 	}
 	// Persist the Participation into the database
 	err = part.Persist()
+	if err != nil {
+		return PersistedParticipation{}, err
```

Comment:
Ideally, return is sufficient here, as in line 182 above. 
However, it is safer not to change the behavior of what is returned in case of an error. 
Use this logic to only call stateProofSecrets.Persist in case of no error, and leave it as is otherwise. 
```
if err == nil {
    err = stateProofSecrets.Persist() 
}
```

Reply:
i updated